### PR TITLE
Set `pool_size` config for `pandas_profiling` to 1 during test execution on Windows

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

The `pipelines-windows` job is flaky and sometimes fails with the following error:

https://github.com/mlflow/mlflow/runs/7494511091?check_suite_focus=true#step:6:154

```
Windows fatal exception: code 0x80000003
Thread 0x000004cc (most recent call first):
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\multiprocessing\pool.py", line 470 in _handle_results
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\threading.py", line 870 in run
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\threading.py", line 926 in _bootstrap_inner
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\threading.py", line 890 in _bootstrap
Thread 0x00002014 (most recent call first):
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\multiprocessing\pool.py", line 422 in _handle_tasks
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\threading.py", line 870 in run
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\threading.py", line 926 in _bootstrap_inner
  File "c:\hostedtoolcache\windows\python\3.7.9\x64\lib\threading.py", line 890 in _bootstrap
```

To fix this issue, set the `pool_size` config for `pandas_profiling` to 1 during test execution on Windows to avoid `multiprocessing`.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Run the `pipelines-windows` job multiple times and all the attempts succeed.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
- https://github.com/mlflow/mlflow/actions/runs/2730645202/attempts/2
- https://github.com/mlflow/mlflow/actions/runs/2730645202/attempts/3
- https://github.com/mlflow/mlflow/actions/runs/2730645202/attempts/4
- https://github.com/mlflow/mlflow/actions/runs/2730645202/attempts/5

**Review by dbczumar — COMMENTED** [review]
@harupy Won't this just mute the root cause of the error, making it more difficult to figure out what happened?

**Comment by harupy** [comment]
@dbczumar You're right :) It look like the error comes from `pandas_profiling`:

I found `pandas_profiling` disables a test with `reason=multiprocessing+pytest broken on Windows`:

https://github.com/ydataai/pandas-profiling/blob/f8bad5dde27e3f87f11ac74fb8966c034bc22db8/tests/unit/test_console.py#L17-L21

```python
@pytest.mark.skipif(os.name == "nt", reason="multiprocessing+pytest broken on Windows")
def test_console_multiprocessing(console_data, test_output_dir):
    report = test_output_dir / "test_samples.html"
    console.main(["-s", "--pool_size", "0", str(console_data), str(report)])
    assert report.exists(), "Report should exist"
```

File: tests/pipelines/conftest.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+@pytest.fixture(autouse=True)
+def patch_get_pool_size():
+    # Set the pool_size config for pandas_profiling to 1 on Windows to avoid using multiprocessing
+    if os.name == "nt":
+        with mock.patch("mlflow.pipelines.utils.step._get_pool_size", return_value=1):
+            yield
+    else:
+        yield
```
Comment (harupy) [inline tests/pipelines/conftest.py:13]: This may not work since an MLP is executed in a subprocess.

**Comment by harupy** [comment]
autoformat

**Comment by harupy** [comment]
autoformat

**Comment by harupy** [comment]
autoformatt

**Review by WeichenXu123 — APPROVED** [review]
LGTM

File: mlflow/pipelines/utils/step.py [hunk 2]
```diff
@@ -81,6 +81,12 @@ def display_html(html_data: str = None, html_file_path: str = None) -> None:
             subprocess.run(["open", html_file_path], check=True)
 
 
+# Prevent pandas_profiling from using multiprocessing on Windows while running tests.
+# multiprocessing and pytest doesn't play well together.
```
Comment (WeichenXu123) [inline mlflow/pipelines/utils/step.py:85]: Could you create a ticket on pandas profiling OSS project ?
Once they address it we can remove this.

Comment (harupy) [reply mlflow/pipelines/utils/step.py:85]: Is this a pandas_profiling issue?

Comment (WeichenXu123) [reply mlflow/pipelines/utils/step.py]: I don't know, but pandas profiling code triggers the error, it might has some relationship

**Comment by harupy** [comment]
Similar issue for matplotlib: https://github.com/matplotlib/matplotlib/issues/20143

**Review by dbczumar — APPROVED** [review]
Brilliant! Thanks @harupy !

File: mlflow/pipelines/utils/step.py [hunk 3]
```diff
@@ -81,6 +81,13 @@ def display_html(html_data: str = None, html_file_path: str = None) -> None:
             subprocess.run(["open", html_file_path], check=True)
 
 
+# Prevent pandas_profiling from using multiprocessing on Windows while running tests.
+# multiprocessing and pytest doesn't play well together on Windows.
```
Comment (harupy) [inline mlflow/pipelines/utils/step.py:85]:
```suggestion
# multiprocessing and pytest don't play well together on Windows.
```

**Pull request merged**